config_src/nuopc_driver/mom_ocean_model_nuopc.F90

```diff
@@ -686,8 +690,13 @@ subroutine ocean_model_restart(OS, timestamp, restartname)
       "restart files can only be created after the buoyancy forcing is applied.")
 
   if (present(restartname)) then
-     call save_restart(OS%dirs%restart_output_dir, OS%Time, OS%grid, &
-          OS%restart_CSp, GV=OS%GV, filename=restartname)
+     if (present(num_rest_files)) then
+       call save_restart(OS%dirs%restart_output_dir, OS%Time, OS%grid, &
```

Comment:
If you pass an input optional argument as an optional argument to another routine, it is present or not in that second routine depending on whether it is present or not in the first.  There is no need for the `present(num_rest_files)` test, and just the code on lines 694 and 695 does the same thing as this entire 7 line block.